### DISCUSSION TITLE
codeMirrorRef and milkdownRef are switched in Demo code

### DISCUSSION THREAD

**Discussion by sytolk** [Milkdown/milkdown#661] [General]
Look at the `<OnlineEditor ref={milkdownRef} content={md} onChange={milkdownListener} />` its have `milkdownRef`
https://github.com/Saul-Mirone/milkdown/blob/main/website/component/Demo/Demo.tsx#L73

but in `milkdownListener` codeMirrorRef  is used: 
https://github.com/Saul-Mirone/milkdown/blob/main/website/component/Demo/Demo.tsx#L52

As the result:
```javascript
if (!current) return;
current.update(markdown); // -> this code will never reached
```
but if I fix this issue executing this `current.update(markdown);` will loop listener for changes here: 
https://github.com/Saul-Mirone/milkdown/blob/main/website/component/MilkdownEditor/editor.ts#L35

**Comment by Saul-Mirone** [comment]
It's by design. It can confirm sync will only happen by one direction. Which means if change comes from codemirror, milkdown will be synced and this sync won't trigger another sync to codemirror, to prevent a infinite loop.
